sdk/src/main/kotlin/io/rover/sdk/streams/Operators.kt

```diff
@@ -95,6 +105,61 @@ internal fun <T, R> Publisher<T>.map(transform: (T) -> R): Publisher<R> {
     }
 }
 
+/**
+ * Retry operator that when given a number of retries will not emit onError calls further downstream but will instead
+ * resubscribe to the publisher that it was previously subscribed to until the attempted number of retries is equal to the
+ * given number of retries, at this point, another onError call from upstream will be emitted downstream.
+ */
+
+internal fun <T> Publisher<T>.retry(numberOfRetries: Int): Publisher<T> {
+    val prior = this
+
+    return Publisher { subscriber ->
+        var attemptedRetries = 0
+        var subscrip: Subscription? = null
+        prior.subscribe(
+            object : Subscriber<T> {
+                override fun onComplete() {
+                    log.w("Retrying retry on complete $attemptedRetries")
+                    subscriber.onComplete()
+                }
+
+                override fun onSubscribe(subscription: Subscription) {
+                    subscrip = subscription
+                    val consumerSubscription = object : Subscription {
+                        override fun cancel() {
+                            subscription.cancel()
+                        }
+
+                        override fun request(n: Long) {
+                            if (n != Long.MAX_VALUE) throw RuntimeException("Backpressure signalling not supported.  Request Long.MAX_VALUE.")
+                            subscription.request(Long.MAX_VALUE)
+                        }
+                    }
+                    subscriber.onSubscribe(consumerSubscription)
```

Comment:
hmmm, I think this will make your downstream subscriber receive a fresh onSubscribe() call every time we retry. I don't think that's quite right.  Maybe only have it do that on the first onSubscribe from the upstream.

Maybe this was the source of the troubles yesterday?